addons/html_editor/static/tests/delete/find_adjacent_position.test.js

```diff
@@ -0,0 +1,228 @@
+import { describe, expect, test } from "@odoo/hoot";
+import { setupEditor } from "../_helpers/editor";
+import { getContent, setSelection } from "../_helpers/selection";
+import { unformat } from "../_helpers/format";
+
+function findAdjacentPosition(editor, direction) {
+    const deletePlugin = editor.plugins.find((p) => p.constructor.name === "delete");
+    const selection = editor.document.getSelection();
+    const { anchorNode, anchorOffset } = selection;
+
+    return deletePlugin.findAdjacentPosition(anchorNode, anchorOffset, direction);
+}
+
+async function testFindPosition({ previous, next, extraSetup }) {
+    const { editor, el } = await setupEditor(previous);
+    extraSetup?.(editor, el);
```

Comment:
if you remove the extrasetup thing (and rename the function), you can make this function more powerful
```js
        test("should find adjacent character", async () => {
            await testFindPosition({
                previous: "<p>a[]bcd</p>",
                next: "<p>ab[]cd</p>",
            });
        });
```
vs
```js
        test("should find adjacent character", async () => {
           const { editor, el } = await setupEditor("<p>a[]bcd</p>");
           // no "extra setup here, but if there was, it could be here
          expectNextPosition("<p>ab[]cd</p>");  // <- synchronous
          // can now add more steps if necessary
        });
```

Reply:
```js
function expectNextPosition(editor, previous, next) {
    let [node, offset] = findAdjacentPosition(editor, "forward");
    setSelection({ anchorNode: node, anchorOffset: offset });
    expect(getContent(editor.el)).toBe(next);

    [node, offset] = findAdjacentPosition(editor, "backward");
    setSelection({ anchorNode: node, anchorOffset: offset });
    expect(getContent(editor.el)).toBe(previous);
}


        test("should find adjacent character", async () => {
            const previous = "<p>a[]bcd</p>";
            const next = "<p>ab[]cd</p>";
            const { editor } = await setupEditor(previous);
            expectNextPosition(editor, previous, next);
        });
```